src/main/java/org/terasology/dynamicCities/minimap/CentreOverlay.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.dynamicCities.minimap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.dynamicCities.settlements.SettlementsCacheComponent;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.geom.Rect2f;
+import org.terasology.math.geom.Rect2fTransformer;
+import org.terasology.math.geom.Rect2i;
+import org.terasology.math.geom.Vector2f;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.minimap.overlays.MinimapOverlay;
+import org.terasology.rendering.assets.texture.Texture;
+import org.terasology.rendering.nui.Canvas;
+import org.terasology.utilities.Assets;
+
+import java.util.Optional;
+
+public class CentreOverlay implements MinimapOverlay {
+
+    private EntityRef settlementCachingEntity;
+    private Logger logger = LoggerFactory.getLogger(DistrictOverlay.class);
+
+    private Vector2f iconSize = new Vector2f(32f, 32f);
```

Comment:
I don't know if this is a big problem, but it would be nice to put 32f in a Constant